### ISSUE TITLE
[local gc] GetLoaderAllocatorObjectForGC always returns null

### ISSUE THREAD

**Issue by davmason** [dotnet/coreclr#17361]
https://github.com/dotnet/coreclr/blob/22ce24ec7f1ea042c56a6531489def81c6fd7b02/src/gc/env/gcenv.object.h#L125-L129

Collectible assemblies won't function without this

**Comment by davmason** [comment]
@Maoni0 I think we'll have to add an interface call for this. When GetLoaderAllocatorObjectForGC is called on a MethodDesc, it has to call in to the Module it was loaded from to get the Allocator. The code path is not straightforward and we can't just grab it from some offset.

**Comment by davmason** [comment]
Actually, I started to poke at this and we build with FEATURE_REDHAWK defined in clrgc.dll, which disables COLLECTIBLE_CLASS, which is what all the collectible stuff is behind. So even if we do this feature collectible assemblies won't work.

**Comment by Maoni0** [comment]
yes we need an interface call to actually get the object if the methodtable indicates it's Collectible.

clrgc.dll should not be built with FEATURE_REDHAWK for coreclr...

**Comment by jkotas** [comment]
> clrgc.dll should not be built with FEATURE_REDHAWK for coreclr...

Tracked by #14701
